msvc/scripts/genproj.cs

```diff
@@ -884,6 +883,19 @@ string groupConditional
 			}
 		).ToList ();
 
+		if (commonFiles.Any (f => f.EndsWith("build\\common\\Consts.cs"))) {
+			var genconstsRelativePath = "$(SolutionDir)\\msvc\\scripts\\genconsts.csproj";
+			result.Append ($"  <ItemGroup>{NewLine}");
+			result.Append ($"    <ProjectReference Include=\"{genconstsRelativePath}\">{NewLine}");
+			result.Append ($"      <Name>genconsts</Name>");
+			result.Append ($"      <Project>{SlnGenerator.genconsts_csproj_guid}</Project>");
```

Comment:
`Name` and `Project` are generally not required by command-line builds, if that's relevant to you here; they could potentially be dropped--but there's no need to do so since you already have the information.